src/terminal/parser/ut_parser/OutputEngineTest.cpp

```diff
@@ -596,6 +596,110 @@ class Microsoft::Console::VirtualTerminal::OutputEngineTest final
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/23/4", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x23, 0x44));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/23/45", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x23, 0x45));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/23/456", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x23, 0x45));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/34/5", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x55));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/34/56", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x56));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/345/67", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x67));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/345/678", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x67));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:123/456/789", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:123/4564/789", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:123/4564/7897", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1231/4564/7897", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"#111", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"#123456", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x56));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"#123456789", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"#123145647897", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"orange", color));
+        VERIFY_ARE_EQUAL(color, RGB(255, 165, 0));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"dark green", color));
+        VERIFY_ARE_EQUAL(color, RGB(0, 100, 0));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"medium sea green", color));
+        VERIFY_ARE_EQUAL(color, RGB(60, 179, 113));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"LightYellow", color));
+        VERIFY_ARE_EQUAL(color, RGB(255, 255, 224));
+
+        // Invalid sequences.
```

Comment:
There are some intensive test cases showing that the code is more robust that before.